### ISSUE TITLE
[Question] : Dynamic Subscription possibility

### ISSUE THREAD

**Issue by dotChris90** [eclipse-iceoryx/iceoryx#1286]
Hi Iceoryx team, 

I have just a theoretical question related to Subscriber creation. 
(It's not urgent and just a question if this is possible or not)

Is it possible to create new Subscriber while the callback of listener is executed? 
So basically - is it possible to create new Subscriber while be in this line of code --> https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_examples/callbacks/ice_callbacks_listener_as_class_member.cpp#L72

Background : i was trying to implement the following situation : 
- subs Process listening to a "registration topic" 
- publisher Process sending a "registration struct" (with participantID and a topic of form {"A", "B", "C" } ) to the "registration topic"
- the callback is executed and topic from "registration struct" is extracted 
- now want to create a new Subscriber connected with the extracted topic 

basically a publisher would register and tell the subscriber on which topic subscriber shall listen. 

Thanks for your time - as i said - totally just question.

**Comment by elfenpiff** [comment]
@dotChris90 

This is actually no problem at all. You can create/destroy ports concurrently without restrictions as long as enough ports are available.

You can also attach/detach them in the listener callback. 

But be aware then you are entering the lair of concurrency where unicorns and pink elephants are flying around ;).
So when you attach ports to the listener, be aware that no port is threadsafe and the listener works in one background thread concurrently on all attached objects. When you then access them from outside of the listener callback while the listener is running you have race conditions, maybe memory synchronization issues and torn reads.

**Comment by dotChris90** [comment]
@elfenpiff  thanks for that information! i will give it a try this week. 

BTW congratulation to you! iceoryx 2.0.0!

**Comment by dotChris90** [comment]
@elfenpiff ok alright! tried it out. i will close the ticket because all worked like expected. thanks for answer.  
^^
and have a good week.
